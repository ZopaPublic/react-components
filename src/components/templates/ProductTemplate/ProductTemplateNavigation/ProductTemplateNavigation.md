### Summary

`<ProductTemplateNavigation>` is used in the `<ProductTemplate />` and shouldn't be used separately. It wraps and places a back and/or next link or custom link component above the title of the page.
