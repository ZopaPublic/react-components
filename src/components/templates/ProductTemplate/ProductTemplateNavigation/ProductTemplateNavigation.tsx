import React, { MouseEventHandler, ReactElement } from 'react';
import styled from 'styled-components';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';
import Text from '../../../atoms/Text/Text';
import Icon from '../../../atoms/Icon/Icon';
import Link from '../../../atoms/Link/Link';
import { colors, grid } from '../../../../constants';

interface ProductTemplateNavigationProps {
  prevStep?: ReactElement | string;
  onBackPressed?: MouseEventHandler<HTMLAnchorElement>;
}

const StyledLink = styled(Link)`
  text-decoration: none;
  display: flex;
  align-items: center;
`;

const Wrapper = styled.div`
  @media (min-width: ${grid.breakpoints.m}px) {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }
`;

const BackText = styled(Text).attrs({ color: colors.greyDark })`
  line-height: 0;
`;

export function ProductTemplateNavigation({ prevStep, onBackPressed }: ProductTemplateNavigationProps) {
  return (
    <Wrapper className="my-4 m:mt-0" data-automation="ZA.ProductTemplateNavigation">
      {!prevStep || typeof prevStep === 'string' ? (
        <StyledLink href={prevStep} aria-label="Back" onClick={onBackPressed}>
          <Icon variant={faChevronLeft} color={colors.greyDark} size="xs" className="mr-2" />
          <BackText>Back</BackText>
        </StyledLink>
      ) : (
        prevStep
      )}
    </Wrapper>
  );
}
