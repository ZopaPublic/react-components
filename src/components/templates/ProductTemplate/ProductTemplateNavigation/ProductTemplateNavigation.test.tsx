import React from 'react';
import { fireEvent, screen, render } from '@testing-library/react';
import { ProductTemplateNavigation } from './ProductTemplateNavigation';

describe('<ProductTemplateNavigation />', () => {
  it('renders with a string prevStep', () => {
    const { container } = render(<ProductTemplateNavigation prevStep="/prevStep" />);
    expect(container).toMatchSnapshot();
  });

  it('renders with a component prevStep', () => {
    const { container } = render(<ProductTemplateNavigation prevStep={<>A custom component</>} />);
    expect(container).toMatchSnapshot();
  });

  it('invokes onBackPressed callback', () => {
    const mockOnBackPressed = jest.fn();
    render(<ProductTemplateNavigation onBackPressed={mockOnBackPressed} />);
    fireEvent.click(screen.getByLabelText('Back'));
    expect(mockOnBackPressed).toHaveBeenCalled();
  });
});
