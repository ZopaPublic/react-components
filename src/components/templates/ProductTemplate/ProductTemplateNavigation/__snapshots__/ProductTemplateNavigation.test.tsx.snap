// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTemplateNavigation /> renders with a component prevStep 1`] = `
@media (min-width:768px) {
  .c0 {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }
}

<div>
  <div
    class="c0 my-4 m:mt-0"
    data-automation="ZA.ProductTemplateNavigation"
  >
    A custom component
  </div>
</div>
`;

exports[`<ProductTemplateNavigation /> renders with a string prevStep 1`] = `
.c3 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #4A545E;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  text-align: inherit;
}

.c1 {
  background-color: transparent;
  font-size: inherit;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  line-height: inherit;
  color: #3B46C4;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  border: none;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  color: auto;
}

.c1:hover,
.c1:active {
  color: #21247F;
}

.c1:hover svg path,
.c1:active svg path {
  fill: #21247F;
}

.c1:hover {
  color: auto;
}

.c1:hover svg path {
  fill: auto;
}

.c1:active {
  color: auto;
}

.c1:active svg path {
  fill: auto;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  line-height: 0;
}

@media (min-width:768px) {
  .c0 {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }
}

<div>
  <div
    class="c0 my-4 m:mt-0"
    data-automation="ZA.ProductTemplateNavigation"
  >
    <a
      aria-label="Back"
      class="c1 c2"
      color="#3B46C4"
      href="/prevStep"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-chevron-left fa-w-10 fa-xs mr-2"
        color="#4A545E"
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
        />
      </svg>
      <span
        class="c3 c4"
        color="#4A545E"
      >
        Back
      </span>
    </a>
  </div>
</div>
`;
